Sources/CocoaLumberjack/DDLog.m

```diff
@@ -705,21 +703,18 @@ - (void)lt_addLogger:(id <DDLogger>)logger level:(DDLogLevel)level {
              @"This method should only be run on the logging thread/queue");
 
     dispatch_queue_t loggerQueue = NULL;
-
     if ([logger respondsToSelector:@selector(loggerQueue)]) {
         // Logger may be providing its own queue
-
-        loggerQueue = [logger loggerQueue];
+        loggerQueue = logger.loggerQueue;
     }
 
     if (loggerQueue == nil) {
         // Automatically create queue for the logger.
         // Use the logger name as the queue name if possible.
-
         const char *loggerQueueName = NULL;
 
         if ([logger respondsToSelector:@selector(loggerName)]) {
-            loggerQueueName = [[logger loggerName] UTF8String];
+            loggerQueueName = logger.loggerName.UTF8String;
```

Comment:
Should we add logger.loggerQueueName property here?

Reply:
Why? What's wrong with having two dots? 😉

Reply:
Also, where would you add that property? You can't add a property with a default implementation on a protocol in ObjC 😉

Reply:
@ffried I could wrap everything in `NSProxy` ... :D
Or we could add static function that get queue name of a logger.
Nothing wrong with two dots, yes, but I would like to distinguish `loggerName property` and `loggerQueueName transform`.
Or it could be done via comment in logger protocol.

```objective-c
// NOTE: We use loggerName .utf8 representation to set loggerQueueName.
// @property (copy, nonatomic, readonly) char *loggerQueueName; // logger.loggerName.UTF8String;
```

Reply:
I don't think that any action is necessary here. We never had a problem with it and it's an internal matter what we do with `loggerName`. Especially since it's an extensible enum, so we basically should always have compile time constants here, which means we also always get an UTF-8 string.

Since we don't need to transform it very often, I think we're fine with what we have now.

Reply:
@ffried 
Just to be sure.
Does `loggerName` allow emoji?

Reply:
Allowed is any string the compiler can process as constant. 
But even emojis are representable in UTF-8 using surrogate pairs:

```objc
#import <Foundation/Foundation.h>

typedef NSString *LoggerName NS_TYPED_EXTENSIBLE_ENUM;

static LoggerName const LoggerNameEmoji = @"🤷‍♂️🧔🏽";

int main(int argc, char *argv[]) {
	@autoreleasepool {
		dispatch_queue_t q = dispatch_queue_create(LoggerNameEmoji.UTF8String, NULL);
		const char *label = dispatch_queue_get_label(q);
		NSString *strLabel = [[NSString alloc] initWithUTF8String:label];
		NSLog(@"%@", strLabel); // -> 2020-01-23 18:27:19.651 Untitled[62409:8550421] 🤷‍♂️🧔🏽
	}
}
```